django/contrib/contenttypes/fields.py

```diff
@@ -236,26 +218,36 @@ def __get__(self, instance, cls=None):
         if instance is None:
             return self
 
+        # We do not want to use ``getattr(instance, self.ct_field)`` here,
+        # because that way we might end up reloading the same ContentType
+        # over and over again (see ticket #5570). Instead, we just get the
+        # content type ID here, and later when we need an actual instance,
+        # use ContentType.objects.get_for_id(), which has a global cache.
```

Comment:
I reworded the entire comment; the more I was looking at it, the less sense it made in the new location; I hope the new wording is clearer.